doc/reference/guidelines.rst

```diff
@@ -0,0 +1,48 @@
+.. highlight:: python
+
+=================
+Coding Guidelines
+=================
+
+Python
+======
+
+- PEP8 ignore E501
+- prefer % over format
+- try to avoid generators and decorators
```

Comment:
Opps i meant decorator. I think the same should applies to generator except that the intellectual price to understand a generator is much lower. (i would say 30s for the decorator to 5s for the generator to 1s for simple loop an return)